### PR TITLE
Add iam key user tables

### PR DESCRIPTION
**Pull request by jplanckeel**
# Context

- Organization_id is optional
- Fix user id in table iam_user

For this PR https://github.com/turbot/steampipe-plugin-scaleway/pull/30 

# Tests

- Test to run a query with organization_id 
- Test to run a query without organization_id

### CODE REVIEW

File: scaleway/table_scaleway_iam_user.go [hunk 1]
```diff
@@ -21,18 +21,18 @@ func tableScalewayIamUser(_ context.Context) *plugin.Table {
...
 		},
 		Get: &plugin.GetConfig{
 			Hydrate:    getIamUser,
-			KeyColumns: plugin.SingleColumn("id"),
+			KeyColumns: plugin.SingleColumn("user_id"),
 		},
 		Columns: []*plugin.Column{
 			{
-				Name:        "id",
+				Name:        "user_id",
```
Comment (cbruno10) [inline scaleway/table_scaleway_iam_user.go:35]: @jplanckeel Why is this being renamed? This would be a breaking change, so we usually deprecate and replace before removing to avoid breaking existing queries.

File: docs/tables/scaleway_iam_api_key.md [hunk 2]
```diff
@@ -2,6 +2,8 @@
 
 API keys allow you to securely connect to scaleway console in your organization.
 
+This table requires an Organization ID to be configured in the scaleway.spc file.
```
Comment (cbruno10) [inline docs/tables/scaleway_iam_api_key.md:5]:
```suggestion
This table requires the `organization_id` config argument to be set.
```

File: docs/tables/scaleway_iam_user.md [hunk 3]
```diff
@@ -2,6 +2,8 @@
 
 Users allow you to connect to scaleway console in your organization.
 
+This table requires an Organization ID to be configured in the scaleway.spc file.
```
Comment (cbruno10) [inline docs/tables/scaleway_iam_user.md:5]:
```suggestion
This table requires the `organization_id` config argument to be set.
```

File: docs/index.md [hunk 4]
```diff
@@ -71,11 +71,12 @@ connection "scaleway" {
   # Set the static credential with the `access_key` and `secret_key` arguments.
   # Alternatively, if no creds passed in config, you may set the environment
   # variables using the `SCW_ACCESS_KEY` and `SCW_SECRET_KEY` arguments.
-  # access_key = "SCWKMH185ZG5THRH7WVX"
-  # secret_key = "ee3b5cb8-2c81-887c-a772-17d46dd34vc7"
-
+  access_key = "YOUR_ACCESS_KEY"
+  secret_key = "YOUR_SECRET_ACCESS_KEY"
+
```
Comment (cbruno10) [inline docs/index.md:76]:
```suggestion
```

File: config/scaleway.spc [hunk 5]
```diff
@@ -8,7 +8,8 @@ connection "scaleway" {
   secret_key = "YOUR_SECRET_ACCESS_KEY"
   
   # Your organization ID is the identifier of your account inside Scaleway infrastructure.
-  organization_id = "YOUR_ORGANIZATION_ID"
+  # This is only required while querying `scaleway_iam_api_key` and `scaleway_iam_user` tables.
```
Comment (cbruno10) [inline config/scaleway.spc:11]:
```suggestion
  # This is only required while querying the `scaleway_iam_api_key` and `scaleway_iam_user` tables. 
```

File: docs/index.md [hunk 6]
```diff
@@ -71,11 +71,12 @@ connection "scaleway" {
...
   # variables using the `SCW_ACCESS_KEY` and `SCW_SECRET_KEY` arguments.
-  # access_key = "SCWKMH185ZG5THRH7WVX"
-  # secret_key = "ee3b5cb8-2c81-887c-a772-17d46dd34vc7"
-
+  access_key = "YOUR_ACCESS_KEY"
+  secret_key = "YOUR_SECRET_ACCESS_KEY"
+  
   # Your organization ID is the identifier of your account inside Scaleway infrastructure.
-  organization_id = "YOUR_ORGANIZATION_ID"
+  # This is only required while querying `scaleway_iam_api_key` and `scaleway_iam_user` tables.
```
Comment (cbruno10) [inline docs/index.md:78]:
```suggestion
  # This is only required while querying the `scaleway_iam_api_key` and `scaleway_iam_user` tables. 
```

File: docs/index.md [hunk 7]
```diff
@@ -71,11 +71,12 @@ connection "scaleway" {
   # Set the static credential with the `access_key` and `secret_key` arguments.
   # Alternatively, if no creds passed in config, you may set the environment
   # variables using the `SCW_ACCESS_KEY` and `SCW_SECRET_KEY` arguments.
-  # access_key = "SCWKMH185ZG5THRH7WVX"
-  # secret_key = "ee3b5cb8-2c81-887c-a772-17d46dd34vc7"
-
+  access_key = "YOUR_ACCESS_KEY"
```
Comment (cbruno10) [inline docs/index.md:74]: Can you please revert these changes? We like to include (fake) examples so users know what values can look like.

**Review by cbruno10 — CHANGES_REQUESTED** [review]
@jplanckeel Please see review comments, thanks!

File: scaleway/table_scaleway_iam_user.go (hunk 1 above)
Comment (jplanckeel) [reply scaleway/table_scaleway_iam_user.go:35]: i change this because this comment in destination PR 

https://github.com/turbot/steampipe-plugin-scaleway/pull/30#discussion_r1302905842

File: docs/index.md (hunk 7 above)
Comment (jplanckeel) [reply docs/index.md:74]: Yes i can revert ;)

File: scaleway/table_scaleway_iam_user.go (hunk 1 above)
Comment (jplanckeel) [reply scaleway/table_scaleway_iam_user.go:35]: We don't have BC because table is not merged in main.

Comment (cbruno10) [reply scaleway/table_scaleway_iam_user.go:35]: @jplanckeel Got it, thanks. I'd recommend then:
- Remove the `KeyColumns` block from the `ListConfig`, since if a user specifies `id` in a query, the plugin will use the `GetConfig` call anyway
- Keep this column name as `id`, since this is the field name we get back from the Scaleway API
- Update the `GetConfig` `KeyColumns` to use `id` again
